### PR TITLE
Update

### LINKED ISSUE #7: Update runtime to 40

### PR DESCRIPTION
**Pull request by bbhtt**
The patch is from https://github.com/endlessm/eos-photos/pull/328, without it, with runtime 42 it refuses to run. With it, it runs and seems to work, but I do get these warnings which seem to be non fatal. I guess if needed we can create similar patches.

```
/app/share/eos-photos/src/endless_photos.py:6: PyGIWarning: GtkClutter was imported without specifying a version first. Use gi.require_version('GtkClutter', '1.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk, Gdk, GLib, GtkClutter, GObject, Gio, Endless
/app/share/eos-photos/src/endless_photos.py:6: PyGIWarning: Endless was imported without specifying a version first. Use gi.require_version('Endless', '0') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk, Gdk, GLib, GtkClutter, GObject, Gio, Endless
/app/share/eos-photos/src/widgets/composite_button.py:38: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  set_value = (my_flag is not 0)
/app/share/eos-photos/src/share/facebook_auth_dialog.py:2: PyGIWarning: WebKit2 was imported without specifying a version first. Use gi.require_version('WebKit2', '4.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk, WebKit2
```

Also while bumping the python dependencies, there was an ever-increasing number of new dependencies being required. I tried adding manifests for each/adding them to original manifests but it became too messy to manage, so I unified all the python deps in one manifest. Let me know which approach is better. Also let me know if anything is present in the runtime and got duplicated, I'll remove it.

Here's the gist:

`python-pyparsing`, `python-packaging` and `python-tomli` were added; these are being required by `setuptools-scm` and/or `Pillow`.

`python3-setuptools-scm-git-archive` is being required by `numpy`; `numpy` had to split from the main manifest as `matplotlib` required it. 

`matplotlib` needs a config file to build now so that was added. Also `qhull` is being required by `matplotlib` for a header file, so that was added. `python3-cppy` is required by `matplotlib`.

`python-pybind11`, `gast`, `ply`, `beniget` and `pythran` are required by `scipy` which had to split from main manifest as `scikit-image` requires it.

`six` was removed as the runtime seems to have it.

I think I couldn't update one or two python deps I forgot which ones but it was requiring wheel and a specific version of `setuptools` and other dependencies but the runtime already has `setuptools 59.8.0`. (added as review comment)

This built locally fine after lots of work, needs testing.

Fixes https://github.com/flathub/com.endlessm.photos/issues/7

### CODE REVIEW

**Comment by bbhtt** [comment]
Hi, @LeandroStanger. Do you still use this? If you can test the above build and let me know if there are any issues it'd be great.

**Comment by wjt** [comment]
Thank you for your contribution here and on https://github.com/endlessm/eos-photos/pull/328! I will try to get these reviewed by someone at Endless. Only last week I was lamenting this app's prehistoric runtime.

**Comment by bbhtt** [comment]
> Thank you for your contribution here and on [endlessm/eos-photos#328](https://github.com/endlessm/eos-photos/pull/328)! I will try to get these reviewed by someone at Endless. Only last week I was lamenting this app's prehistoric runtime.

Hey, thank you. I didn't author that patch so I shouldn't get the credit for it :) I did come up with the same solution while searching the error message online, but then I saw that there was a PR already and decided to use Andrej's patch.

**Comment by wjt** [comment]
Did you generate the Python manifest by hand or with [flatpak-pip-generator](https://github.com/flatpak/flatpak-builder-tools/tree/master/pip) ?

**Comment by bbhtt** [comment]
> Did you generate the Python manifest by hand or with [flatpak-pip-generator](https://github.com/flatpak/flatpak-builder-tools/tree/master/pip) ?

Most of it was already there, I just moved it to one manifest, manually bumped their versions and added the new dependencies as they were being required during the build.

File: com.endlessm.photos.json [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-    "runtime-version": "3.36",
+    "runtime-version": "42",
     "sdk": "org.gnome.Sdk",
     "command": "eos-photos",
     "finish-args": [
-        "--share=ipc", "--share=network",
-        "--socket=x11", "--socket=wayland",
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
```
Comment (bbhtt) [inline com.endlessm.photos.json]: I think we can change this to `fallback-x11`.

File: python-requirements.json [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+        "sources": [
+            {
+                "type": "archive",
+                "url": "https://github.com/scipy/scipy/releases/download/v1.8.1/scipy-1.8.1.tar.gz",
+                "sha256": "9e3fb1b0e896f14a85aa9a28d5f755daaeeb54c897b746df7a55ccb02b340f33"
+            }
+        ]
+	},
+	{
+        "name": "python3-PyWavelets",
```
Comment (bbhtt) [inline python-requirements.json:344]: Couldn't update this one to latest release, it was somehow failing

File: python-requirements.json [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+        "sources": [
+            {
+                "type": "file",
+                "url": "https://files.pythonhosted.org/packages/17/6b/ef989cfb3acff2ea966c5f28a7443ccaec2ee136675dfa0366db2635f78a/PyWavelets-1.1.1.tar.gz",
+                "sha256": "1a64b40f6acb4ffbaccce0545d7fc641744f95351f62e4c6aaa40549326008c9"
+            }
+        ]
+	},
+	{
+        "name": "python3-scikit-image",
```
Comment (bbhtt) [inline python-requirements.json:358]: This one as well

File: com.endlessm.photos.json (hunk 1 above)
Comment (GeorgesStavracas) [reply com.endlessm.photos.json]: Yes, please change to `fallback-x11`

File: python-requirements.json (hunk 3 above)
Comment (GeorgesStavracas) [reply python-requirements.json:358]: That's okay

**Review by GeorgesStavracas — APPROVED** [review]
Besides the `fallback-x11` change, this generally looks good to me. We can land this after that change. At some point we'll need to issue another eos-photos release based on this, and potentially ported to GTK4, but that's for the future.

**Comment by bbhtt** [comment]
Are the appstream patches needed? https://github.com/endlessm/eos-photos/commit/b69e029f393eabd1dbcbfcc9011ecec057011886 and https://github.com/endlessm/eos-photos/commit/f8b124d25ad847071d6faefe6b655c533b4c8a42 Or maybe we can switch eos-photos from the tag to a git commit, that'd allow dropping the GTK patch I added, since it has been merged already.

**Comment by wjt** [comment]
On https://github.com/flathub/flathub/issues/3238 I requested more Endlessers with commit access to this repo.

Thank you @bbhtt for your contribution here!

**Pull request merged**